### PR TITLE
buttons added

### PR DESCRIPTION
**Pull request by cfoskin**
JIra: https://issues.jboss.org/browse/AEROGEAR-7590

Added the buttons to for the resolvers. This is basically the code from https://github.com/pb82/data-sync-ui/tree/resolvers/
I have just brought over what is needed to get this piece working and modified it to use buttons (which do nothing right now)

some screenshots of how it looks:
screen shot 2018-07-17 at 17 07 18

screen shot 2018-07-17 at 17 06 37

@pb82 as you can see this is mostly code from your branch.

### CODE REVIEW

**Comment by pb82** [comment]
@cfoskin Looks great, one thing: `Add Resolver` is aligned right while `Edit/Delete` is centered. Is there a reason for this?

**Comment by cfoskin** [comment]
@pb82  yes I just noticed that also. I will adjust this

**Comment by pb82** [comment]
@cfoskin I would also hide (in terms of CSS) the cross in the header rows. It serves no purpose other than collapsing the row which is already possible by clicking on the row. And it messes with the layout on smaller screens.

File: ui/components/schema/TypeList.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+} from "patternfly-react";
+
+import style from "./structureView.css";
+import GetResolvers from "../../graphql/GetResolvers.graphql";
+
+class TypeList extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
```
Comment (pb82) [inline ui/components/schema/TypeList.js]: If there's no default state we can remove this.

File: ui/components/schema/TypeList.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        console.log("selected resolver");
+        console.log(resolver.responseMapping);
+    }
+
+    createResolver(fieldName) {
+        console.log(fieldName);
+    }
+
+    renderLoading() {
+        return <div>Loading</div>;
```
Comment (pb82) [inline ui/components/schema/TypeList.js]: We can use @josemigallas 's Spinner here

File: ui/components/schema/structureView.css [hunk 3]
```diff
@@ -50,3 +50,24 @@
 .structure-item-row {
     margin-top: 5px;
 }
+.flexWrapper {
```
Comment (pb82) [inline ui/components/schema/structureView.css]: Why is this css added to the structure view? This looks duplicate to https://github.com/aerogear/data-sync-ui/blob/master/ui/components/schema/schemaContainer.css

File: ui/graphql/GetResolvers.graphql [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    field
+    responseMapping
+    requestMapping
+    DataSource {
+        id
+        name
+        type
+    }
+  }
+}
```
Comment (pb82) [inline ui/graphql/GetResolvers.graphql]: eslint requires a newline at the end of each file

File: ui/components/schema/TypeList.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    }
+
+    renderResolverForField(name, data) {
+        const { resolvers } = data;
+        const resolver = resolvers.find(item => item.field === name);
+        // const resolver = { }; just here to test the buttons look ok
+        if (resolver) {
+            return (
+                <div style={{ textAlign: "center" }}>
+                    <div className="btn-group">
```
Comment (josemigallas) [inline ui/components/schema/TypeList.js]: I'd give these buttons a little bit of separation.

File: ui/components/schema/structureView.css (hunk 3 above)
Comment (josemigallas) [reply ui/components/schema/structureView.css]: These styles are not used anywhere in `StructureView.js`.

File: ui/graphql/GetResolvers.graphql [hunk 6]
```diff
@@ -0,0 +1,14 @@
+query GetResolvers($schemaId: Int!, $type: String!) {
```
Comment (josemigallas) [inline ui/graphql/GetResolvers.graphql]: Use agreed indentation of 4 spaces please.

File: ui/components/schema/StructureView.js [hunk 7]
```diff
@@ -20,77 +22,26 @@ const wellKnownTypes = [
...
     const { types } = compiled.data.__schema;
     const relevantTypes = types.filter(type => wellKnownTypes.indexOf(type.name) < 0);
 
     return (
         <ListView>
             {
-                relevantTypes.map((type, index) => renderType(type, index))
+                relevantTypes.map((type, index) => (
+                    <TypeList
+                        key={index}
```
[the remark this answers is not in the document]
Comment (pb82) [reply ui/components/schema/StructureView.js]: `type.name` should work here

**Comment by cfoskin** [comment]
Splitting the Buttons was a pain and I could not get any property of the Patternfly Button to make it work so open to better way of doing it.

**Comment by pb82** [comment]
@cfoskin I believe you could use CSS margin to split the buttons

**Comment by cfoskin** [comment]
@pb82 Ive tried with css to put space between them but they keep jumping to the new line. Maybe you can see what is wrong? I added the css here https://github.com/aerogear/data-sync-ui/pull/46/commits/a43d18033e4e3f6c4a7425b3e17ef8ca9167c329

**Comment by cfoskin** [comment]
@pb82  ok to merge this one now? 
screen shot 2018-07-19 at 10 52 24

**Comment by pb82** [comment]
@cfoskin  It's still using a div instead of the spinner for the loading state but i think we can merge it and take care of that later. @josemigallas are you ok with merging?

**Comment by cfoskin** [comment]
@pb82 I think I updated that: https://github.com/cfoskin/data-sync-ui/blob/88a05d00750c81a38f7c072e62895633902864c3/ui/components/schema/SchemaContainer.js#L96

**Comment by cfoskin** [comment]
Merging as spinner added and will unblock testing of modal architecture work

**Pull request merged**